plugins/aws-appmesh/plugin/commands.go

```diff
@@ -302,8 +323,27 @@ func (plugin *Plugin) deleteIngressRules(
 
 // deleteEgressRules deletes the iptable rules for egress traffic.
 func (plugin *Plugin) deleteEgressRules(iptable *iptables.IPTables) error {
+
+	// Check if egress chain exist before deleting
+	exists := false
+	chains, err := iptable.ListChains("nat")
+	if err != nil {
+		return log.Errorf("failed to list chains: %v", err)
```

Comment:
I'm not sure if this is wrong, but I'll note this is _slightly_ semantically different than the rest of the error calls in this file.

Most errors are handled like this:

```go
log.Errorf("Failed to parse netconfig from args: %v.", err)
return err
```

This will print a formatted message, then return the underlying error. Returning the `log.Errorf` will instead return a new error with the original error's message embedded in it.

For the most part I could only imagine this mattering if there's some detailed error inspection upstream of this return, which I doubt is the case, but might be worth double-checking.

Reply:
Will change that, good to keep code consistent!